flang/lib/Parser/message.cpp

```diff
@@ -272,19 +272,51 @@ static llvm::raw_ostream::Colors PrefixColor(Severity severity) {
   return llvm::raw_ostream::SAVEDCOLOR;
 }
 
+// TODO: Make these configurable, based on verbosity level.
+static constexpr int MAX_CONTEXTS_EMITTED{2};
+static constexpr bool OMIT_SHARED_CONTEXTS{true};
+
 void Message::Emit(llvm::raw_ostream &o, const AllCookedSources &allCooked,
     bool echoSourceLine) const {
   std::optional<ProvenanceRange> provenanceRange{GetProvenanceRange(allCooked)};
   const AllSources &sources{allCooked.allSources()};
   sources.EmitMessage(o, provenanceRange, ToString(), Prefix(severity()),
       PrefixColor(severity()), echoSourceLine);
+  // Always refers to whether the attachment in the loop below is a context.
```

Comment:
I think that `attachmentIsContext_` here refers to `*this`, not to its first attachment.  Things would be more clear if you omitted the `isContext` variable and just used `attachment->attachmentIsContext_` below, or moved this Boolean flag variable into that loop and initialized it to `attachment->attachmentIsContext_` on each iteration.

Reply:
https://github.com/llvm/llvm-project/blob/480202f0d16f7dbc5c650aea6e8dfd9eca5b999d/flang/include/flang/Parser/message.h#L273